### PR TITLE
Pakiti client: manage the configuration

### PR DESCRIPTION
**Pull request by jouvin**

### CODE REVIEW

File: features/pakiti/client/config_schema.pan [hunk 1]
```diff
@@ -0,0 +1,11 @@
+unique template features/pakiti/client/config_schema;
+
+
+type pakiti_client_config = {
+    "curl"     : string = "curl"
+    "encrypt"  ? string
+    "expect"   : string = "200 OK"
+    "rndsleep" : long = 7200
+    "site"     : string
+    "url"      : string
```
Comment (jrha) [inline features/pakiti/client/config_schema.pan:10]:
```suggestion
    "curl"     : string_trimmed = "curl"
    "encrypt"  ? string_trimmed
    "expect"   : string_trimmed = "200 OK"
    "rndsleep" : long(0..86400) = 7200
    "site"     : string_trimmed
    "url"      : type_hostURI
```

File: features/pakiti/client/config.pan [hunk 2]
```diff
@@ -6,34 +6,87 @@ include PAKITI_RPMS;
...
+        PAKITI_SERVER_CERTIFICATION_CHAIN_SOURCE_DEFAULT;
+    } else {
+        undef;
+    };
+};
 
+# Build configuration file
+include 'components/metaconfig/config';
+include 'features/pakiti/client/config_schema';
+prefix '/software/components/metaconfig/services/{/etc/pakiti/pakiti-client.conf}';
```
Comment (jrha) [inline features/pakiti/client/config.pan:35]: If this is for Pakiti v3 can we make this:
```suggestion
prefix '/software/components/metaconfig/services/{/etc/pakiti3/pakiti-client.conf}';
```

File: features/pakiti/client/config.pan [hunk 3]
```diff
@@ -6,34 +6,87 @@ include PAKITI_RPMS;
...
+    if ( file_exists(PAKITI_SERVER_PUB_KEY_SOURCE) ) {
+        SELF['encrypt'] = PAKITI_SERVER_PUB_KEY_FILE;
+    } else if ( PAKITI_SERVER_PROTOCOL != 'https' ) {
+        error('Pakiti server pub key is missing and feed protocol is not https');
+    };
+    SELF['rndsleep'] = PAKITI_CLIENT_SLEEP;
+    SELF;
+};
+'convert/yesno' = true;
+bind '/software/components/metaconfig/services/{/etc/pakiti/pakiti-client.conf}/contents' = pakiti_client_config;
```
Comment (jrha) [inline features/pakiti/client/config.pan:61]: If this is for Pakiti v3 can we make this:
```suggestion
bind '/software/components/metaconfig/services/{/etc/pakiti3/pakiti-client.conf}/contents' = pakiti_client_config;
```

File: features/pakiti/client/config.pan [hunk 4]
```diff
@@ -6,34 +6,87 @@ include PAKITI_RPMS;
...
-    push(dict("name", "pakiti_update",
-            "user", "root",
-            "frequency", PAKITI_CLIENT_UPDATE_FREQ,
-            'command', '/usr/bin/pakiti-client --conf ' + PAKITI_CLIENT_CONF));
+"/software/components/cron/entries" = {
+    SELF[length(SELF)] = dict(
+        "name", "pakiti_update",
+        "user", "root",
+        "frequency", PAKITI_CLIENT_UPDATE_FREQ,
+        "command", "/usr/bin/pakiti-client --conf /etc/pakiti/pakiti-client.conf",
```
Comment (jrha) [inline features/pakiti/client/config.pan:89]: If this is for Pakiti v3 can we make this:
```suggestion
        "command", "/usr/bin/pakiti-client --conf /etc/pakiti3/pakiti-client.conf",
```

File: features/pakiti/client/config_schema.pan (hunk 1 above)
Comment (jouvin) [reply features/pakiti/client/config_schema.pan:10]: Done

File: features/pakiti/client/config.pan (hunk 3 above)
Comment (jouvin) [reply features/pakiti/client/config.pan:61]: Done

File: features/pakiti/client/config.pan (hunk 4 above)
Comment (jouvin) [reply features/pakiti/client/config.pan:89]: Done

**Comment by jrha** [comment]
Fair enough, we run both a fork of Pakiti2 and Pakiti3, but we can deal with that.
If you rebase this panlint should be able to run now.

**Comment by jrha** [comment]
Well, it's a use-case which will hopefully disappear soon, so I don't think it's worth worrying about too much.

**Pull request merged**